spyder/plugins/completion/plugin.py

```diff
@@ -381,12 +383,33 @@ def on_mainwindow_visible(self):
             provider_info['instance'].on_mainwindow_visible()
 
     # ---------------------------- Status bar widgets -------------------------
-    def register_statusbar_widgets(self):
+    def register_statusbar_widgets(self, plugin_loaded=True):
         """Register status bar widgets for all providers with the container."""
-        container = self.get_container()
         for provider_key in self.providers:
-            provider = self.providers[provider_key]['instance']
-            container.register_statusbar_widgets(provider.STATUS_BAR_CLASSES)
+            provider_on = self.get_conf(
+                ('enabled_providers', provider_key), True)
+            if provider_on:
+                self.register_statusbar_widget(
+                    provider_key, plugin_loaded=plugin_loaded)
+
+    def register_statusbar_widget(self, provider_name, plugin_loaded=True):
+        """Register the given statusbar widget."""
```

Comment:
I think this docstring is incorrect because it doesn't receive any widgets to register.

Perhaps a more accurate description would be

    Register statusbar widgets associated to a provider.

Also, please document what `provider_name` and `plugin_loaded` do in the docstring.